### ISSUE TITLE
GitHub API has request number limitation

### ISSUE THREAD

**Issue by simonzhaoms** [mlhubber/mlhub#80]
Currently the quotas for GitHub API request is 60 per hour.  It is not a problem for general use.  But it will become a problem for testing, otherwise, I won't find this problem.

See https://developer.github.com/v3/#rate-limiting

When reaching the limitation, the following will be obtained when opening `https://api.github.com/repos/gjwgit/animate/contents/MLHUB.yaml` by browser:

```json
{
  "message": "API rate limit exceeded for 167.220.255.151. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

Below is what will be obtained by command:

```console
$ wget -O abc https://api.github.com/repos/gjwgit/animate/contents/MLHUB.yaml
--2019-01-02 16:07:57--  https://api.github.com/repos/gjwgit/animate/contents/MLHUB.yaml
Resolving api.github.com (api.github.com)... 192.30.255.117, 192.30.255.116
Connecting to api.github.com (api.github.com)|192.30.255.117|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2019-01-02 16:07:58 ERROR 403: Forbidden.
```

The remaining quota for request can be found by:

```console
$ curl -i https://api.github.com/users/octocat
HTTP/1.1 403 Forbidden
Date: Wed, 02 Jan 2019 08:10:17 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 259
Server: GitHub.com
Status: 403 Forbidden
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 0
X-RateLimit-Reset: 1546418069
X-GitHub-Media-Type: github.v3; format=json
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
X-GitHub-Request-Id: 950F:8344:B049E64:DD6FA45:5C2C71E9

{
  "message": "API rate limit exceeded for 167.220.255.151. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

**Comment by simonzhaoms** [comment]
This problem was found when I tested code for generating a curated list of mlhub packages into `Packages.yaml`.  Since there are 14 packges now, I will use up 60 requests after I run 5 times the script to generate `Packages.yaml` from packages' `MLHUB.yaml` on their GitHub repo.

**Comment by simonzhaoms** [comment]
Solved in commit 4520e5d
